third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -1113,32 +1217,61 @@ pub(crate) struct Resolver<'a> {
     loader: &'a Loader,
     module_store: &'a ModuleStorageAdapter,
     binary: BinaryType,
+
+    module_storage: &'a dyn ModuleStorageV2,
 }
 
 impl<'a> Resolver<'a> {
-    fn for_module(
+    pub(crate) fn new_for_function_in_module(
+        function: &Function,
         loader: &'a Loader,
         module_store: &'a ModuleStorageAdapter,
-        module: Arc<Module>,
-    ) -> Self {
+        module_storage: &'a dyn ModuleStorageV2,
+    ) -> PartialVMResult<Self> {
+        let module_id = function
+            .module_id()
+            .ok_or_else(|| {
```

Comment:
we could move this inside module_id call, with error saying "Function {} must have a module owner", i.e. not always, but when we call, it should.

Reply:
Hmmm, yes, but scripts still do not have it is not really an error, there a few debug use cases where we optionally print things. My take is that the function should not be owned, but `LoadedFunction` should be because this is the one that has full context, and can actually carry arced module or script. @runtian-zhou, what do you think?

I have tried that approach, and it requires changing types from Arc<Function> to LoadedFunction throughout, and I was not convinced this is better than just special casing for scripts.